### PR TITLE
[Bug] Fix CSS on `page-main` in mobile layouts

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
Noticed two bugs which are fixed by this PR:
- `page-main` not overflowing correctly in mobile mode
- `card-link` not extending properly in mobile mode

@l0uden - we should probably have some screenshot tests for the mobile layouts as well. So far I've been manually checking every week as I know there are currently other priorities, but would be good to add this to the next sprint 🙏 

## Screenshot
**Before:**
**`page-main` not overflowing correctly in mobile mode**

**`card-link` not extending properly in mobile mode**
Screenshot 2024-02-26 at 10 08 27

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/changelog.d/20240226_100408_huong_li_nguyen_fix_css_grid.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+- A bullet item for the Deprecated category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX ([#1](https://github.com/mckinsey/vizro/pull/1))
+
+-->
+
+### Fixed
+
+- Fix CSS bug on `page-main` overflowing `page-container` in mobile layouts. ([#331](https://github.com/mckinsey/vizro/pull/331))
+
+<!--
```
Comment (petar-qb) [inline vizro-core/changelog.d/20240226_100408_huong_li_nguyen_fix_css_grid.md:42]: Should we also mention a fix for card-link not extending properly here?

**Review by petar-qb — APPROVED** [review]
This looks good!

File: vizro-core/changelog.d/20240226_100408_huong_li_nguyen_fix_css_grid.md (hunk 1 above)
Comment (huong-li-nguyen) [reply vizro-core/changelog.d/20240226_100408_huong_li_nguyen_fix_css_grid.md:42]: No, all good! It was introduced by a PR that has been merged but it hasn't been released yet, so the user is not aware of that issue yet :)

Comment (petar-qb) [reply vizro-core/changelog.d/20240226_100408_huong_li_nguyen_fix_css_grid.md:42]: Makes sense then. 😄

**Review by antonymilne — APPROVED** [review]

**Pull request merged**